### PR TITLE
refactor: address all issue in #14

### PR DESCRIPTION
**Pull request by sandangel**
I have fixed all issue as describe in the comment.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { INITIAL_STATE_KEY } from './keys';
-import { Action } from '@ngrx/store';
 
-export function Store<TState>(initialState?: TState): (target: Function) => void;
-export function Store(initialState?: any): (target: Function) => void;
-export function Store(initialState: any = {}) {
+export function Store<TState extends object>(initialState?: TState): (target: Function) => void;
```
Comment (albyrock87) [inline src/store.ts]: I don't think that this is a good idea. A feature state can also be a number. Can we go back to `any`?

File: src/select.ts [hunk 2]
```diff
@@ -18,22 +17,23 @@ export function Select<TState = any, TValue = any>(
...
-      if (paths.length) {
-        selectorOrFeature = [selectorOrFeature, ...paths].join('.');
+      let propsArray = [selectorOrFeature, ...paths];
+      if (!paths.length) {
+        propsArray = selectorOrFeature.split('.');
       }
-      fn = fastPropGetter(selectorOrFeature);
-    } else {
+      fn = fastPropGetter(propsArray);
+    } else if (typeof selectorOrFeature === 'function') {
```
Comment (albyrock87) [inline src/select.ts]: I don't think we need this check: if it's not a `string` than it's a `Selector<TState, TValue>`.

File: src/select.ts [hunk 3]
```diff
@@ -18,22 +17,23 @@ export function Select<TState = any, TValue = any>(
...
     // Nothing here? Use propery name as selector
-    if (typeof selectorOrFeature === 'undefined') {
+    if (!selectorOrFeature) {
       selectorOrFeature = name;
     }
     // Handle string vs Selector<TState, TValue>
     if (typeof selectorOrFeature === 'string') {
-      if (paths.length) {
-        selectorOrFeature = [selectorOrFeature, ...paths].join('.');
+      let propsArray = [selectorOrFeature, ...paths];
```
Comment (albyrock87) [inline src/select.ts]: What about `const propsArray = paths.length ? [selectorOrFeature, ...paths] : selectorOrFeature.split('.');` ?

File: src/spec/index.spec.ts [hunk 4]
```diff
@@ -157,7 +157,7 @@ describe('actions', () => {
   });
 
   it('selects sub state', () => {
-    const state: {
+    const featureState: {
```
Comment (albyrock87) [inline src/spec/index.spec.ts]: Can you change this in `globalState`? The feature state is the `myFeature` property.

**Review by albyrock87 — CHANGES_REQUESTED** [review]
Can you change those things?
Thank you for your contribution.

File: src/store.ts (hunk 1 above)
Comment (sandangel) [reply src/store.ts]: sorry, I did not know that

**Review by albyrock87 — APPROVED** [review]

**Pull request merged**